tools/tune/eq/eq_norm.m

```diff
@@ -40,34 +40,37 @@
 m_max_iir = max(eq.iir_eq_db);
 sens_fir = sum(m_lin_fir.*w_lin)/sum(w_lin);
 sens_iir = sum(m_lin_iir.*w_lin)/sum(w_lin);
-g_offs = 10^(eq.norm_offs_db/20);
+g_offs_iir = 10^(eq.iir_norm_offs_db/20);
+g_offs_fir = 10^(eq.fir_norm_offs_db/20);
 
 %% Determine scaling gain
-switch lower(eq.norm_type)
+switch lower(eq.iir_norm_type)
         case 'loudness'
-                g_fir = 1/sens_fir;
                 g_iir = 1/sens_iir;
         case '1k'
-                g_fir = 1/m_lin_fir(i1k);
                 g_iir = 1/m_lin_iir(i1k);
-
         case 'peak'
-                g_fir = 10^(-m_max_fir/20);
                 g_iir = 10^(-m_max_iir/20);
         otherwise
-                error('Requested normalization is not supported');
+                error('Requested IIR normalization is not supported');
+end
+switch lower(eq.fir_norm_type)
+        case 'loudness'
```

Comment:
The scaling factor calculations for FIR and IIR do not appear to account for possible zero values, which may result in division by zero (1/sens_iir or 1/m_lin_iir(i1k)). You might want to add a tick to avoid this.

Example: 

```
>> eq.f = 48e3;
      i1k = find(eq.f > 1e3, 1, 'first') - 1;
      disp(i1k)
     0

```
then
```
        if sens_fir ~= 0
            g_fir = 1/sens_fir;
        else
            error('Sensation level is zero for FIR, cannot normalize by loudness.');
        end
```
For other's as well